client/src/legacy/GridField.js

```diff
@@ -495,6 +501,100 @@ $.entwine('ss', function($) {
     }
   });
 
+  $('.js-injector-boot .grid-field .grid-field__search-holder').entwine({
+    Component: null,
+
+    onmatch() {
+      this._super();
+
+      // Make sure this appears at the top of the gridfield
+      this.prependTo(this.parent());
+
+      const cmsContent = this.closest('.cms-content').attr('id');
+      const context = (cmsContent)
+        ? { context: cmsContent }
+        : {};
+
+      const Search = loadComponent('Search', context);
+      this.setComponent(Search);
+
+      this.refresh();
+    },
+
+    onunmatch() {
+      this._super();
+      // solves errors given by ReactDOM "no matched root found" error.
+      const container = this[0];
+      if (container) {
+        ReactDOM.unmountComponentAtNode(container);
+      }
+    },
+
+    close() {
+      const props = this.data('schema');
+
+      const ajaxData = [{
+        name: props.clearAction,
+        value: '',
+        filter: 'hidden',
+        triggerChange: false
+      }];
+
+      this.getGridField().reload({ data: ajaxData });
+    },
+
+    search(data) {
+      const props = this.data('schema');
+
+      const ajaxData = [{
+        name: props.searchAction,
+        value: '',
+        filter: 'show',
+        triggerChange: false
+      }];
+
+      for (const [key, value] of Object.entries(data)) {
+        const name = `filter[${props.gridfield}][${key}]`
+        ajaxData.push({ name, value });
+      }
+
+      this.getGridField().reload({ data: ajaxData });
+    },
+
+    refresh() {
+      const props = this.data('schema');
+      const Search = this.getComponent();
+      const handleHide = () => this.close();
+      const handleSearch = (data) => this.search(data);
+
+      ReactDOM.render(
+        <Search
+          id={`${props.gridfield}Search`}
+          className="no-change-track"
+          identifier={`${props.gridfield}Search`}
+          display="VISIBLE"
+          displayBehavior="HIDEABLE"
+          onHide={handleHide}
+          onSearch={handleSearch}
+          {...props}
+        />,
+        this[0]
+      );
+    },
+  });
+
+  /**
+   * Stop enter from submitting the whole 'page form' and
+   * allow the search component to handle it
+   */
+  $('.js-injector-boot .grid-field .search-box__content-field').entwine({
+    onkeydown: function(e) {
```

Comment:
That can probably be fixed on the Search/SearchBox component directly. Otherwise, we'll have to fix it everywhere else.